scripts/lint_prowjobs/main.go

```diff
@@ -0,0 +1,316 @@
+		if presubmitConfig.JobBase.Spec.ServiceAccountName != jc.ServiceAccountName {
+			return false, findLineNumber(fileContentsString, "serviceaccountName:"), fmt.Sprintf(`Incorrect service account configuration, please configure service account as => serviceaccountName: %s`, jc.ServiceAccountName)
+		}
+		return true, 0, ""
+	})
+}
+
+func PresubmitMakeTargetCheck(jc *JobConstants) presubmitCheck {
+	return presubmitCheck(func(presubmitConfig config.Presubmit, fileContentsString string) (bool, int, string) {
+		if strings.Contains(presubmitConfig.JobBase.Name, "lint") {
+			return true, 0, ""
+		}
+		jobMakeTargetMatches := regexp.MustCompile(`make (\w+) .*`).FindStringSubmatch(strings.Join(presubmitConfig.JobBase.Spec.Containers[0].Command, " "))
+		jobMakeTarget := jobMakeTargetMatches[len(jobMakeTargetMatches)-1]
+		makeCommandLineNo := findLineNumber(fileContentsString, "make")
+		if strings.Contains(presubmitConfig.JobBase.Name, "cni") && jobMakeTarget != jc.CniMakeTarget {
+			return false, makeCommandLineNo, fmt.Sprintf(`Invalid make target, please use the "%s" target`, jc.CniMakeTarget)
+		} else if strings.Contains(presubmitConfig.JobBase.Name, "helm-chart") && jobMakeTarget != jc.HelmMakeTarget {
+			return false, makeCommandLineNo, fmt.Sprintf(`Invalid make target, please use the "%s" target`, jc.HelmMakeTarget)
+		} else if strings.Contains(presubmitConfig.JobBase.Name, "release-tooling") && jobMakeTarget != jc.ReleaseToolingMakeTarget {
+			return false, makeCommandLineNo, fmt.Sprintf(`Invalid make target, please use the "%s" target`, jc.ReleaseToolingMakeTarget)
+		} else if jobMakeTarget != jc.DefaultMakeTarget {
+			return false, makeCommandLineNo, fmt.Sprintf(`Invalid make target, please use the "%s" target`, jc.DefaultMakeTarget)
+		}
+		return true, 0, ""
+	})
+}
+
+func PostsubmitMakeTargetCheck(jc *JobConstants) postsubmitCheck {
+	return postsubmitCheck(func(postsubmitConfig config.Postsubmit, fileContentsString string) (bool, int, string) {
+		jobMakeTargetMatches := regexp.MustCompile(`make (\w+) .*`).FindStringSubmatch(strings.Join(postsubmitConfig.JobBase.Spec.Containers[0].Command, " "))
+		jobMakeTarget := jobMakeTargetMatches[len(jobMakeTargetMatches)-1]
+		makeCommandLineNo := findLineNumber(fileContentsString, "make")
+		if jobMakeTarget != jc.DefaultMakeTarget {
+			return false, makeCommandLineNo, fmt.Sprintf(`Invalid make target, please use the "%s" target`, jc.DefaultMakeTarget)
+		}
+		return true, 0, ""
+	})
+}
+
+func getFilesChanged(gitRoot string, pullBaseSha string, pullPullSha string) ([]string, []string, error) {
+	presubmitFiles := []string{}
+	postsubmitFiles := []string{}
+	gitDiffCommand := []string{"git", "-C", gitRoot, "diff", "--name-only", pullBaseSha, pullPullSha}
+	fmt.Println("\n", strings.Join(gitDiffCommand, " "))
+
+	gitDiffOutput, err := exec.Command("git", gitDiffCommand[1:]...).Output()
```

Comment:
You need to handle this error

Reply:
I am returning the error from the helper function and handling it in the main() function [here](https://github.com/aws/eks-distro-prow-jobs/blob/8e95ab774f4cc2f363baf754c3aa9ac8be25a03f/scripts/lint_prowjobs/main.go#L251). Is that acceptable?